---
title: Variant Descriptions
---

# Spike Protein

Spike proteins are proteins involved in viral attachment, fusion and entry. This protein is found in many viruses and is a target of interest for anti-viral drugs and vaccines.

## Analysis of Variants


1. Val367Phe : RBD
   1. Asp398Tyr : RBD
3. Ala522Val : RBD
4. Lys921Gln : Salt Bridge Between HR1 and HR2



### Receptor Binding Domain (RBD)

The RBD domain of coronaviruses is part of the Spike protein and is involved in binding with host proteins. Studies suggest that the SARS-Cov-2 RBD domain bounds to human and bat Angiotensin-Converting Enzyme 2 (ACE2) receptors. Research shown that the affinity of the Spike protein and the ACE2 receptor is key to the susceptibility of host infection by SARS-CoV [[1]](#ref1).
The RBD domain contains a core stucture and a Receptor Binding Domain (RBM). The RBM binds to the ACE2 structure. The RBD domain of SARS-2-CoV has been shown to be ancestral to both human and bat SARS-CoVs and to share 73%\~76% of the ABD and 50\~53% of the RBM with them. This are higher similarities than with the human and bat MERS and MERS-like viruses. The RBM contains 14 residues in contact with ACE2 (between position 437 and 508). 9 of them are fully conserved with human, civet and bat SARS-CoV and 4 are partially conserved [[1]](#ref1). No variant observed in our variant analyses are impacting the RBM, but 3 of them impact the RBD on either sides of the RBM.

| |
|:--------:|
| <img src="RBD_structure.jpg " alt="drawing" width="700"/> |
| **Figure 1** : SARS-Cov - spike protein RBD Structure [[6]](#ref6) |




| |
|:--------:|
| <img src="ACE2_RBD_structure.jpg " alt="drawing" width="700"/> |
| **Figure 2** : SARS-Cov - Important residues for ACE2 recognition and species specificity [[6]](#ref6) |


##### Asp398Tyr

A G-to-T variation causes a Asp398Tyr substitution in the position 398 of the S protein, in the RBD somain. This residue is reported fully conserved across MERS and SARS coronaviruses [[2]](#ref2). This could indicate a counter selection on variations at this position. It is, however, located in a putative Beta sheet structure, and research report that mutations impacting the binding of RBM to ACE2 are the one located in loops or turns [[3]](#ref3). This study is based on 32 mutations induced in the RBD. It indicates that residues impacting the affinity of the Spike protein with ACE2 include positively and negatively charged, polar and hydrophobic residues, indicating an electrostatic and hydrophobic component in the binding RBM-ACE2.
The same research shows that mutations that affect the binding ability of RBM are more numerous than the ones that do not. This result encourages us to investigate more closely variants affecting the RBD. In addition, this variant causes a change in charge and/or polarity, that could impact the interactions between RBD and ACE2.

##### Ala522Val

A C-to-T variation in second position of the codon causes an Ala522Val subsitution in the RBD of the spike protein. This variation does not cause a change in hydrophobic or electrostatic properties of the spike protein, since both Alanine and Valine are hydrophobic amino acids.

### Heptad Repeat (HR) Domains

Heptad Repeat Domains are regions containing repeated motifs of seven amino acids. The exact amino acid is rarely important in the repetition, but the nature of the amino acid is. These motifs create a tube structure with an hydrophobic core. In spike proteins HR1 and HR2 motifs create salt bridges. They form a tertiary structure involved in the docking and fusion of the virus membrane with the host cell.



| |
|:--------:|
| <img src="S_structure_HR.jpg " alt="drawing" width="700"/> |
| **Figure 3** : SARS-Cov - Structure of the spike protein and Heptad repeat domains. Overall structure: determinants of the hydrophobic core formation and central ions interactions. (A and B) Ribbon diagrams colored light and dark grey for the HR1 and HR2 polypeptides, respectively. Pink spheres on the central 3-fold axis indicate the chloride ions. In A, the axes of the HR1 helices in the trimer are drawn as green tubes, highlighting the two stutters in red. The coiled coil axis is dark red (vertical at the trimer center). The helical turns are numbered from N- to C-terminus for one of the subunits (black and white numbers are used for HR1 and HR2, respectively). The N- and C-terminal ends of the model are indicated for one HR1/HR2 heterodimer. The 2 columns between the vertical scale bar on the left and the ribbon diagram indicate the residues and the 3- and 4-residue repeat pattern of the side chains facing the 3-fold axis of the coiled coil. Black and red fonts indicate polar and non-polar side chains, respectively. Residues within green boxes are strictly conserved. Red boxes in the second column highlight the stutters. In B, the side chains of polar residues within the hydrophobic core are drawn in green and labeled. Water molecules are indicated as small red spheres. Inter chain salt bridges (1 and 2, labeled in blue, corresponding to Lys 903 to Glu 1188 and Lys 929 to Glu 1163, respectively) are also indicated (basic side chains are in blue, acidic in red). The central cavity is displayed as a gold surface. (C) Slab of the model viewed down the 3-fold axis to show the chloride ions. Pink arrows indicate the center of the slab in panel B: top and bottom panels display chloride ions 1 and 2, respectively. The hydrogen bonding network propagating from the central ions toward the outside—which highly constrains the HR2 main chain, is indicated. Several of the Asn and Gln residues labeled are part of the Asn/Gln zipper (see text). As a guide for orientation, the axes of the 3 HR1 α-helices are drawn and labeled in green. The top panel is a view from below the atom, and the bottom panel from above it, relative to panel 2B. (D) Helical wheel after correction for the stutters (as in the register line of Fig. 1B). HR1 left, HR2 middle: as in panel A, polar and non-polar side chains are black and red, respectively (notice the strong amphipathic character of the two helices). Positions a and d are highlighted within a circle with a red background. The right panel shows a diagram of the interactions in the 6-helix bundle. [[5]](#ref5) |



##### Lys921Gln : Salt Bridge


An A-to-C variant causes a Lys921Gln amino acid replacement in the spike glycoprotein (S) .  S is a homotrimeric protein containing two subdomain S1 and S2 mediating receptor recognition and membrane fusion [[4]](#ref4). The S2 subdomains contain two heptad repeat (repeats of units containing seven amino acids) regions: HR1 and HR2. The substitution we observed is located in HR1. It aligns with residue 903 of the SARS spike protein that forms a salt bridge with Gln1188 with HR2. This salt bridge is involved in the formation of a HR1/HR2 hairpin structures [[5]](#ref5). This site invariably contains Lys in all human SARS-related coronaviruses as well as in many other coronaviruses. However, more distantly related coronaviruses including transmissible gastroenteritis coronavirus (TGEV), the porcine respiratory coronavirus (PRCV), the canine coronavirus (CCV), the feline peritonitis virus (FIPV), and the porcine epidemic diarrhea virus (PDEV) all contain Gln at the corresponding position (Figure 4. [[5]](#ref5)). In these spike proteins, the salt bridge involves lysine in position 900 instead. SARS-Cov-2 has no lysine at that position. The Lys921Gln change would then prevent the formation of the salt bridge with Gln1188 and may have structural and functional implications for the spike protein structure and, consequently, COVID-19 virulence.



| |
|:--------:|
| <img src="HR_align.jpg " alt="drawing" width="700"/> |
| **Figure 4** :The HR1 and HR2 regions in the coronavirus S glycoprotein. (A) Linear diagram indicating the relative locations of the segments described in the text. The S1 and S2 regions are labeled. FP: putative fusion peptide region. TM: transmembrane region. SP: signal peptide. (B) Sequence alignments of the HR1 and HR2 regions of coronavirus spike proteins of the three different groups (G1 to G3) and the unclassified SARS coronavirus (blue arrow) using ClustalW (Thompson et al., 1994). See the Supplementary material for the complete names of the viruses used. The numbers at the top line correspond to the SARS amino acid sequence, the structure of which is described in the text. The alignment is color-coded according to sequence conservation: red, strictly conserved; green, highly conserved; blue, conserved; black, variable. The alignment in the HR2N region was manually modified to match the structural superposition with the corresponding region of the MHV protein (PDB accession numbers 1WDF and 1WDG, Xu et al., 2004c). Residues with ordered electron density have a grey background in the SARS line. The cloned fragment contained all residues between the boxed columns (further highlighted with a vertical empty arrow below the sequence). Two additional lines at the bottom summarize the following: the “Register” line provides the abcdefg heptad repeat assignment with letters in black for the residues actually observed in a helical conformation in the structure, with the two HR1 stutters in grey background. Note the insertion of exactly two heptad repeats, both in HR1 and HR2, in the S protein of group 1 coronaviruses. The “Interaction” line shows the two salt bridges (1 and 2, see Fig. 2B) in a yellow background. In the case of HR1, this line provides also the residues participating in the asparagine/glutamine zipper shown in Fig. 3 (labeled “N”), the knob-into-hole interactions with either partner within the trimer (labeled B or C), and the residues lining the central cavity shown in Fig. 2B (labeled with a star). Residues forming salt bridges 1 and 2 connecting HR1 with HR2 have a yellow background, with a number below to indicate the partner in each chain. Note that salt bridge 1 is conserved, but is sometimes made with HR1 residue 900 (from the previous helical turn) instead of 903. The structure shows that the side chain of residue 900 can contact the HR2 1188 side chain equally well. Blue background columns indicate HR1 residues interacting with the two central chloride ions. Vertical grey background columns identify HR2 residues in the extended segments (HR2N and HR2C) that pack their side chains into hydrophobic pockets in the HR1 interhelical grooves. [[6]](#ref6) |



##### Ala942Ser : Hydrogen bond

A G-to-T variation causes a Ala942Ser subsitution in the HR1 domain of the Spike protein. This residue align with a residue involved in an hydrogen bond in MERS virus. If this bond is present in SARS-COV-2, it could be disrupted by this variation since it replaces an hydrophobic amino acid to a polar one.

##### Lys921Gln : Salt Bridge

### References


<a name="ref1"> **[1]** </a>
Wan, Yushun, et al. "Receptor recognition by the novel coronavirus from Wuhan: an analysis based on decade-long structural studies of SARS coronavirus." Journal of virology 94.7 (2020).

<a name="ref2"> **[2]** </a>
Tai, Wanbo, et al. _"Characterization of the receptor-binding domain (RBD) of 2019 novel coronavirus: implication for development of RBD protein as a viral attachment inhibitor and vaccine."_ Cellular & Molecular Immunology (2020): 1-8.

<a name="ref3"> **[3]** </a>
Chakraborti, Samitabh, et al. _"The SARS coronavirus S glycoprotein receptor binding domain: fine mapping and functional characterization."_ Virology journal 2.1 (2005): 73.

<a name="ref4"> **[4]** </a>
Walls AC, Tortorici MA, Bosch B-J, Frenz B, Rottier PJM, DiMaio F, Rey FA, Veesler D. _Cryo-electron microscopy structure of a coronavirus spike glycoprotein trimer._ Nature. 2016 Mar 3;531(7592):114–117. PMCID: PMC5018210

<a name="ref5"> **[5]** </a>
Duquerroy S, Vigouroux A, Rottier PJM, Rey FA, Bosch BJ. _Central ions and lateral asparagine/glutamine zippers stabilize the post-fusion hairpin conformation of the SARS coronavirus spike glycoprotein._ Virology. Elsevier; 2005 May 10;335(2):276–285. PMID: 15840526

<a name="ref6"> **[6]** </a>
Li, Fang, et al. "Structure of SARS coronavirus spike receptor-binding domain complexed with receptor." Science 309.5742 (2005): 1864-1868.



# Coordinate conversion

S/S_coordinates_conversion.tab -> better

| Accession | Source | Description |
|:------:|:-----------:|:-----------|
| NC_045512 | [NC_045512.2](https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2)  | Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1 (SARS-2-Cov) |
| FJ647223.1 | [FJ647223.1](https://www.ncbi.nlm.nih.gov/nuccore/FJ647223) ) |  Murine coronavirus MHV-1 |
| NC_038294.1 | [NC_038294.1](https://www.ncbi.nlm.nih.gov/nuccore/NC_038294.1) | Betacoronavirus England 1 |
| NC_004718.3 | [NC_004718.3](https://www.ncbi.nlm.nih.gov/nuccore/NC_004718.3) | SARS coronavirus |
| FJ647218.1 | [FJ647218.1](https://www.ncbi.nlm.nih.gov/nuccore/FJ647218.1) | Murine coronavirus RA59/R13 |
| MN996527.1 | [MN996527.1](https://www.ncbi.nlm.nih.gov/nuccore/MN996527.1) | Severe acute respiratory syndrome coronavirus 2 isolate WIV02 |
| NC_028806.1 | [NC_028806.1](https://www.ncbi.nlm.nih.gov/nuccore/NC_028806.1)  | Swine enteric coronavirus strain Italy/213306/2009  |
| AY714217.1 | [AY714217.1](https://www.ncbi.nlm.nih.gov/nuccore/AY714217.1)  | SARS Coronavirus CDC#200301157 |
| KF367457.1 | [KF367457.1](https://www.ncbi.nlm.nih.gov/nuccore/KF367457.1)  | Bat SARS-like coronavirus WIV1 |
| NC_038861.1 | [NC_038861.1](https://www.ncbi.nlm.nih.gov/nuccore/NC_038861.1)  | Transmissible gastroenteritis virus |
| NC_002306.3 | [NC_002306.3](https://www.ncbi.nlm.nih.gov/nuccore/NC_002306.3) | Feline infectious peritonitis virus |
| MG772934.1 | [MG772934.1](https://www.ncbi.nlm.nih.gov/nuccore/MG772934.1) | Bat SARS-like coronavirus isolate bat-SL-CoVZXC21 |
| AY463059.1 | [AY463059.1](https://www.ncbi.nlm.nih.gov/nuccore/AY463059) | SARS coronavirus ShanghaiQXC1 |
| MN996528.1 | [MN996528.1](https://www.ncbi.nlm.nih.gov/nuccore/MN996528.1) | Severe acute respiratory syndrome coronavirus 2 isolate WIV04 |
| DQ022305.2  | [DQ022305.2](https://www.ncbi.nlm.nih.gov/nuccore/DQ022305.2) |  Bat SARS coronavirus HKU3-1 |
| JX504050  | [JX504050.1](https://www.ncbi.nlm.nih.gov/nuccore/JX504050.1) |  Human coronavirus NL63 isolate NL63/RPTEC/2004 |
